thoth/storages/graph/postgres.py

```diff
@@ -4822,6 +4894,41 @@ def sync_revsolver_result(self, document: Dict[str, Any]) -> None:
                     platform=entry["platform"],
                 )
 
+    def _check_package_solved(self, session: Session, package_name: str, package_version: str,
+                              package_index: str) -> Tuple:
+        """We check if the package has been solved before syncing it to the database."""
+        # Check if the package has a known index
+        if package_index:
+            python_package_version = (
+                session.query(PythonPackageVersion)
+                .join(PythonPackageIndex)
+                .filter(PythonPackageVersion.package_name == package_name)
+                .filter(PythonPackageVersion.package_version == package_version)
+                .filter(PythonPackageIndex.url == package_index if package_index else None)
+                .first()
+            )
+        else:
+            raise PythonIndexNotProvided(
+                f"Trying to sync package {package_name!r} in version {package_version!r} "
+                "which does not have corresponding Python entity record"
+            )
+        # Check if we run the solver for a specific package already
+        if python_package_version:
+            python_package_version = self._create_python_package_version(
+                session,
+                package_name=package_name,
+                package_version=package_version,
+                index_url=package_index,
+                sync_only_entity=False,
+            )
+        else:
+            raise SolverNotRun(
+                f"Trying to sync package {package_name!r} in version {package_version!r} "
+                f"not solved by any solver."
+            )
+        # We return python_package_entity id and python_package_version id if solver has been run.
+        return python_package_version.entity_id, python_package_version.id
```

Comment:
```
            python_package_version = (
                session.query(PythonPackageVersion)
                .join(PythonPackageIndex)
                .filter(PythonPackageVersion.package_name == package_name)
                .filter(PythonPackageVersion.package_version == package_version)
                .filter(PythonPackageIndex.url == package_index if package_index else None)
                .first()
            )

        # Check if we run any solver for a specific package already
        if not python_package_version:
            raise SolverNotRun(
                f"Trying to sync package {package_name!r} in version {package_version!r} "
                f"not solved by any solver."
            )

        # We return python_package_entity id and python_package_version id if any solver has been run.
        return python_package_version.entity_id, python_package_version.id
```

Reply:
No need to check index either, as we schedule solver only for known index. We just want to know if that package has been solved from any solver, one is enough :)

Reply:
Thanks changed it :+1: